src/path.cr

```diff
@@ -647,7 +647,18 @@ struct Path
   # Path.windows("foo\\bar").join(Path.posix("baz/baq")) # => Path.windows("foo\\bar\\baz/baq")
   # ```
   def join(parts : Enumerable) : Path
-    new_name = String.build do |str|
+    if parts.is_a?(Indexable)
+      # If we know how many parts we have we can compute an approximation of
+      # the string's capacity: this path's size plus the parts' size plus the
+      # separators between them
+      capacity = @name.bytesize +
+                 parts.sum { |part| (part.is_a?(Path) ? part.@name : part.to_s).bytesize } +
```

Comment:
Use `Path#to_s`, which returns `@name`, instead of `part.@name`.
```suggestion
                 parts.sum(&.to_s.bytesize) + parts.size
```

Reply:
Good idea, thanks!